src/utils/math.hpp

```diff
@@ -249,32 +143,19 @@ constexpr unsigned int count_leading_zeros_impl(N n, std::size_t w) {
  * @see count_leading_ones()
  */
 template<typename N>
-constexpr unsigned int count_leading_zeros(N n) {
-#if defined(__GNUC__) || defined(__clang__)
-	// GCC’s `__builtin_clz` returns an undefined value when called with 0
-	// as argument.
-	// [<http://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html>]
-	if (n == 0) {
-		// Return the quantity of zero bits in `n` rather than
-		// returning that undefined value.
-		return static_cast<unsigned int>(bit_width(n));
+constexpr std::enable_if_t<std::is_integral_v<N>, unsigned int> count_leading_zeros(N n) {
+	const auto x = static_cast<std::make_unsigned_t<N>>(n);
+	constexpr auto mask = decltype(x){1};
```

Comment:
In this type of scenario, I would avoid `auto`.

```cpp
constexpr decltype(x) mask{1};
```

Reply:
Seconded.

Reply:
Changed.